local/compose/create.go

```diff
@@ -602,32 +644,48 @@ func getAliases(s types.ServiceConfig, c *types.ServiceNetworkConfig) []string {
 	return aliases
 }
 
-func getNetworkMode(p *types.Project, service types.ServiceConfig) container.NetworkMode {
+func getNetworkMode(ctx context.Context, c client.APIClient, p *types.Project, service types.ServiceConfig) (container.NetworkMode, error) {
 	mode := service.NetworkMode
 	if mode == "" {
 		if len(p.Networks) > 0 {
 			for name := range getNetworksForService(service) {
-				return container.NetworkMode(p.Networks[name].Name)
+				return container.NetworkMode(p.Networks[name].Name), nil
 			}
 		}
-		return container.NetworkMode("none")
+		return container.NetworkMode("none"), nil
 	}
 
-	// FIXME incomplete implementation
+	serviceName := ""
 	if strings.HasPrefix(mode, "service:") {
-		panic("Not yet implemented")
-	}
-	if strings.HasPrefix(mode, "container:") {
-		panic("Not yet implemented")
+		_, _ = fmt.Sscanf(mode, "service:%s", &serviceName)
+		serviceContainers, err := c.ContainerList(ctx, moby.ContainerListOptions{
```

Comment:
seems to me the need to lookup containers while we just did by caller methods applying convergence strategy demonstrates a design issue. If we can pass moby API client, we could also pass observed state with known containers.